### PR TITLE
Bump etcd to v3.1.0 GA

### PR DESCRIPTION
**Pull request by smarterclayton**
Had to workaround GRPC incompatibilities with CRI and rkt, neither of which we are actually using in 1.5.

[test]

### CODE REVIEW

**Comment by smarterclayton** [comment]
Just a flake, @liggitt as requested.  See justification for the two hacks.

File: vendor/k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime/api.pb.go [hunk 1]
```diff
@@ -2960,7 +2960,7 @@ var _ grpc.ClientConn
 
 // This is a compile-time assertion to ensure that this generated file
 // is compatible with the grpc package it is being compiled against.
-const _ = grpc.SupportPackageIsVersion3
+const _ = grpc.SupportPackageIsVersion4
```
Comment (liggitt) [inline vendor/k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime/api.pb.go]: how is this part of kube using a different version of grpc than us?

Comment (smarterclayton) [inline vendor/k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime/api.pb.go]: Kube is still on etcd 3.1.0-rc1 track.

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
[merge]

**Comment by smarterclayton** [comment]
As per Jordan's verbal OK and the desire to align with what we are going to ship sooner rather than later.

**Pull request merged**